qemu/tests/virtio_serial_various_chardev_hotplug.py

```diff
@@ -0,0 +1,73 @@
+from virttest import error_context
+from virttest.utils_misc import normalize_data_size
+from qemu.tests.virtio_console import add_chardev
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Hot-plug Various chardev
+
+    1. Boot a guest without chardev, no serial port & no pci
+    2. Hot plug Unix chardev backend
+    3. Hot plug udp chardev backend
+    4. Hot plug null backend
+    5. Hot plug file backend
+    6. Hot plug pty backend
+    7. Hot plug ringbuf backend
+    8. Write data to ringbuf
+    9. Read data from ringbuf
+    10. Hot-unplug Unix chardev backend
+    11. Hot-unplug udp chardev backend
+    12. Hot-unplug null backend
+    13. Hot-unplug file backend
+    14. Hot-unplug pty backend
+    15. Hot-unplug ringbuf backend
+
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def cmd_qmp_log(vm, cmd, args):
+        reply = vm.monitor.cmd_qmp(cmd, args)
+        if "error" in reply:
+            if reply["error"]["class"] == "CommandNotFound":
+                test.cancel("qmp command %s not supported" % cmd)
```

Comment:
You want to cancel the whole test when it does not support `ringbuf-write` or `ringbuf-read`? I think it does no prevent us testing other device type.

Reply:
> You want to cancel the whole test when it does not support `ringbuf-write` or `ringbuf-read`? I think it does no prevent us testing other device type.

Since this case won't pass it would be better to check the command at first. Or do you think how to deal with the error better?

Reply:
@nanliu-r `try...except...pass`

Reply:
> @nanliu-r `try...except...pass`

still think we shouldn't pass the  error `CommandNotFound`......, how about using `test.error`?

Reply:
@nanliu-r It's up to you, but this way I think it's meaningless to distinguish the error type, we will see it in the log anyway.

Reply:
> @nanliu-r It's up to you, but this way I think it's meaningless to distinguish the error type, we will see it in the log anyway.

we can keep it for more intuitive in the test result.